### ISSUE TITLE
Sequence() works on Compute Capability 6.1, but doesn't work on 8.6

### ISSUE THREAD

**Issue by Delcior** [fynv/ThrustRTC#24]
Trying out the example code from the Thrust Quick start guide: 
```python
import ThrustRTC as trtc

X = trtc.device_vector('int32_t', 10)
Y = trtc.device_vector('int32_t', 10)
Z = trtc.device_vector('int32_t', 10)

# initialize X to 0,1,2,3, ....
trtc.Sequence(X)
```
I get expected result on a machine with compute capability 6.1 (checked with https://gist.github.com/f0k/63a664160d016a491b2cbea15913d549) but it fails on machine with CC=8.6 with the following error
```
Traceback (most recent call last):
  File "/home/test/test.py", line 8, in <module>
    trtc.Sequence(X)
  File "/usr/local/lib/python3.9/dist-packages/ThrustRTC/Transformations.py", line 34, in Sequence
    check_i(native.n_sequence(vec.m_cptr, cptr_init, cptr_step))
  File "/usr/local/lib/python3.9/dist-packages/ThrustRTC/Native.py", line 16, in check_i
    raise SystemError("An internal error happend")
SystemError: An internal error happend
```
Some other examples from the Quick Start Guide work on this machine. What other information should I provide?

Thanks for help.

**Comment by fynv** [comment]
What's the CUDA (nvrtc) version? Could it be too low for the device?

**Comment by fynv** [comment]
Another thing is, is there any message from std::out? Strange if not.

**Comment by fynv** [comment]
Ok.. A very likely cause is that internal/cuda_occupancy.h is out-dated. It comes from CUDA 10, so it doesn't know anything about cc 8.x. I'll try replacing it with the one from CUDA 11.6.

**Comment by fynv** [comment]
Please try v0.3.18 to see if it is fixed.

**Comment by Delcior** [comment]
Thank you for quick reply! I've tried new version and looks like it works perfectly, but I spotted another thing which is DeprecationWaring for np.bool. Executing following code:

```Python
# filename: test.py
import ThrustRTC as trtc
import numpy as np

def test():
    vector= trtc.device_vector_from_numpy(np.array([0,1,0], dtype=bool))
    assert True
```
with `pytest test.py`   (numpy==1.21.0 and ThrustRTC==0.3.18) I get:
```
================================================= test session starts ==================================================
platform linux -- Python 3.9.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/test
collected 1 item                                                                                                       

test.py .                                                                                                        [100%]

=================================================== warnings summary ===================================================
test.py::test
  /usr/local/lib/python3.9/dist-packages/ThrustRTC/DVVector.py:93: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    elif nparr.dtype == np.bool:

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================= 1 passed, 1 warning in 0.21s =============================================
```
Tests that are using ThrustRTC run with `-We` flag and this warning causes them to fail. 
Do you need any other informations? (I will try to reply quicker this time)
Thanks for help again.

**Comment by fynv** [comment]
I've just done a fix (v0.3.19).

**Comment by Delcior** [comment]
Thanks! This solved the problem.
